components/Common/OramaSearchBar/index.tsx

```diff
@@ -0,0 +1,91 @@
+import { create, load, search } from '@orama/orama';
+import { useTheme } from 'next-themes';
+import { useCallback, useMemo, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { OramaLogoDark, OramaLogoLight } from './OramaLogo';
+import styles from './index.module.scss';
+import SearchBar from '../SearchBar';
+
+import type { Document, Orama, RawData, Schema } from '@orama/orama';
+import type { FC } from 'react';
+import type { SearchFunction, SearchResult } from '../../../types';
+
+type DocumentMapper = (doc: Document) => SearchResult;
+
+type FooterProps = {
+  theme?: string;
+};
+
+type OramaSearchBarProps = {
+  schema: Schema;
+  index: RawData;
+  documentMapper?: DocumentMapper;
+};
+
+function identityDocumentMapper(doc: Document): SearchResult {
+  return doc as unknown as SearchResult;
+}
+
+async function performSearch(
```

Comment:
This should be on a dedicated place just for search logic, like an util or just a separate file within `OramaSearchBar`

Reply:
See above :)